### PR TITLE
Derelict Shuttle Ruins - Custom shuttle persistance (Reborn edition)

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

This PR aims at bringing back a PR done by @PowerfulBacon (https://github.com/BeeStation/BeeStation-Hornet/pull/5367).

All you need to know about this is in the link, so check it out!

## Why It's Good For The Game

Hard work can persist across multiple rounds, new ruin types that are truely random and you never know what you might come across, and hey, maybe with the supercruise, you can scrap some shuttles you come across for... fun!

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Soon, I'm still sorting out how to put the PR's details down, it's still in draft after all!

</details>

## Changelog
:cl:
add: Adds in derelict shuttle ruins, custom shuttles from previous rounds
code: Significantly improves the map save proc.
config: Adds in new config options related to shuttle saving.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
This didn't actually rely on cross-server comms, it uses local files for storage

**Comment by Crossedfall** [comment]
reloading checks

**Comment by PowerfulBacon** [comment]
Does saving ignore holographic items?

**Comment by PowerfulBacon** [comment]
This basically worked when last testmerged, so I'm fine to put it on again.
There are some things that need consideration:

 - Changes to typepaths / things being deleted that are saved.
 - APCs creating millions of terminals

File: config/config.txt [hunk 1]
```diff
@@ -549,6 +549,20 @@ RESPECT_GLOBAL_BANS
...
+SHUTTLE_RUIN_FILEPATH data/customShuttleSaves/
+## Max filesize in kB that shuttles are allowed to take up on server
+## Default: 1 megabyte
+SHUTTLE_TOTAL_FILESIZE_MAX 1000
+## Max filesize a single shuttle is allowed to take up
+## Default: 40 kB (40,000 characters)
+SHUTTLE_FILESIZE_MAX 40
+## Max amount of shuttles that can spawn at roundstart by SSorbits
+ROUNDSTART_ABANDONED_RUINS 5
+
```
Comment (PowerfulBacon) [inline config/config.txt:565]: (This was never actually removed from config, so config changes don't need to be applied.)

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
We need to check all map files and ensure that there are no invalid typepaths in any of the maps. Any maps that have invalid paths need to be deleted.
This should be done on init.

File: code/modules/clothing/shoes/colour.dm [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+/obj/item/clothing/shoes/sneakers/orange/MouseDrop(atom/over)
+	var/mob/m = usr
+	if(ishuman(m))
+		var/mob/living/carbon/human/hummie = m
+		if(hummie.shoes == src && chained)
+			to_chat(hummie, "<span class='warning'>You start taking off your [src]!</span>")
+			if(!do_after(hummie,15 SECONDS, src))
+				return FALSE
+	return ..()
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]: This is weird

File: code/__DEFINES/flags.dm [hunk 3]
```diff
@@ -36,6 +36,7 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define UNPAINTABLE_1 				(1<<17)
 /// Does the supermatter skip over this atom?
 #define SUPERMATTER_IGNORES_1 (1 << 18) //set this to 18 because tg has some other flags appearantly too if that gets ever ported fix this !!!!
+#define SAVE_SAFE_1					(1<<17)		//! Marks an item as being safe to save. Anything ridiculous or will not work if spawned on a map should not be given this flag.
```
Comment (PowerfulBacon) [inline code/__DEFINES/flags.dm]: This is invalid, needs to be 1<<19

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Some things I noticed.
This needs a script to check that flags aren't being overriden accidentally for review.

File: code/modules/clothing/shoes/colour.dm (hunk 2 above)
Comment (PestoVerde322) [reply code/modules/clothing/shoes/colour.dm]: This is very likely a leftover from the pre-gags era.

**Comment by PestoVerde322** [comment]
Reminder that this needs strong TMing before being merged.

**Comment by PowerfulBacon** [comment]
What are we doing about the old maps that contain outdated things in them?

**Comment by PowerfulBacon** [comment]
time for this to die again :pensive:

**Pull request closed without merging**